src/algebra/monoid_algebra.lean

```diff
@@ -667,6 +702,31 @@ f.single_mul_apply_aux r _ _ _ $ λ a, by rw [zero_add]
 
 end misc_theorems
 
+end add_monoid_algebra
+
+/-!
+#### Conversions between `add_monoid_algebra` and `monoid_algebra`
+
+While we were not able to define `add_monoid_algebra k G = monoid_algebra k (multiplicative G)` due
+to definitional inconveniences, we can still show the types are isomorphic.
+-/
+
+/-- The equivalence between `add_monoid_algebra` and `monoid_algebra` in terms of `multiplicative` -/
+protected def add_monoid_algebra.to_multiplicative [semiring k] [add_monoid G] :
```

Comment:
Should these be `simps`?

Reply:
I've no idea. Do you think they should be? `finsupp.domain_congr` isn't `simps`, so there'd be little point.

Reply:
#4663 is now merged :tada: 
So do you think it's worth it now? My intuition would say yes...

Reply:
Same problem as in our other conversation - #4663 does not teach simps about `add_equiv` and `mul_equiv`.

I think we need a PR that applies #4663 to all of the `_hom` and `_equiv` types. Only then should we worry about attaching `simps` to as many of them as possiblem.

Reply:
I've opened #4818 to get the ball moving on this, but I don't think this PR needs to wait for that.

Reply:
Thanks!